php/php.editor/src/org/netbeans/modules/php/editor/elements/MethodElementImpl.java

```diff
@@ -121,10 +126,66 @@ public static MethodElement createMagicMethod(final TypeElement type, String met
         return retval;
     }
 
+    public static MethodElement createMagicMethod(final TypeElement type, String methodName, int flags, List<Pair<String, String>> arguments, String returnType) {
+        MethodElement retval = new MethodElementImpl(
+                type,
+                methodName,
+                true,
+                0,
+                flags,
+                type.getFilenameUrl(),
+                null,
+                BaseFunctionElementSupport.ParametersImpl.create(fromParameterNames(arguments)),
+                BaseFunctionElementSupport.ReturnTypesImpl.create(returnType),
+                type.isDeprecated());
+        return retval;
+    }
+
+    static String getValidType(@NullAllowed String declaredType, @NullAllowed PhpVersion phpVersion) {
+        if (declaredType == null) {
+            return CodeUtils.EMPTY_STRING;
+        }
+        String type = declaredType.trim();
+        if (phpVersion != null) {
+            switch (type) {
+                case Type.MIXED:
+                    return phpVersion.hasMixedType() ? declaredType : CodeUtils.EMPTY_STRING;
+                case Type.OBJECT:
+                    return phpVersion.hasObjectType() ? declaredType : CodeUtils.EMPTY_STRING;
+                case Type.VOID:
+                    return phpVersion.hasVoidReturnType() ? declaredType : CodeUtils.EMPTY_STRING;
+                default:
+                    break;
```

Comment:
Don't we want here any logging and/or `assert`? For future, if a new `Type` is added.

Reply:
This switch statement handle `String`. In this case, the others(e.g. `array`, `string`, etc.) are allowed. I'll add `// noop` or `return declaredType;` here instead of `assert`.

Reply:
Changed: https://github.com/apache/netbeans/compare/bbbdf487c44ab998427e0e5def72016f69c59d1d..73b894c09ae8792a17f5fbb413732203b80a1496

Reply:
I see, thanks.